internal/mode/static/state/graph/route_common_test.go

```diff
@@ -1788,7 +1789,505 @@ func TestAllowedRouteType(t *testing.T) {
+							Reason:  "NotAllowedByListeners",
+							Message: "Route is not allowed by any listener",
+						},
+					},
+				},
+			},
+			expectedGatewayListeners: []*Listener{
+				createModifiedListener("listener-443", func(l *Listener) {
+					l.Source.AllowedRoutes = &gatewayv1.AllowedRoutes{
+						Namespaces: &gatewayv1.RouteNamespaces{From: helpers.GetPointer(
+							gatewayv1.FromNamespaces("Same"),
+						)},
+					}
+				}),
+			},
+			name: "route not allowed by listener",
+		},
+		{
+			route: createNormalRoute(gw),
+			gateway: &Gateway{
+				Source: gw,
+				Valid:  true,
+				Listeners: []*Listener{
+					createModifiedListener("listener-443", func(l *Listener) {
+						l.Valid = false
+					}),
+				},
+			},
+			expectedSectionNameRefs: []ParentRef{
+				{
+					Idx:         0,
+					Gateway:     client.ObjectKeyFromObject(gw),
+					SectionName: tr.Spec.ParentRefs[0].SectionName,
+					Attachment: &ParentRefAttachmentStatus{
+						AcceptedHostnames: map[string][]string{
+							"listener-443": {"foo.example.com"},
+						},
+						Attached: true,
+					},
+				},
+			},
+			expectedGatewayListeners: []*Listener{
+				createModifiedListener("listener-443", func(l *Listener) {
+					l.Valid = false
+					r := createNormalRoute(gw)
+					r.Conditions = append(r.Conditions, staticConds.NewRouteInvalidListener())
+					r.ParentRefs = []ParentRef{
+						{
+							Idx:         0,
+							Gateway:     client.ObjectKeyFromObject(gw),
+							SectionName: tr.Spec.ParentRefs[0].SectionName,
+							Attachment: &ParentRefAttachmentStatus{
+								AcceptedHostnames: map[string][]string{
+									"listener-443": {"foo.example.com"},
+								},
+								Attached: true,
+							},
+						},
+					}
+					l.L4Routes = map[L4RouteKey]*L4Route{
+						CreateRouteKeyL4(tr): r,
+					}
+				}),
+			},
+			expectedConditions: []conditions.Condition{staticConds.NewRouteInvalidListener()},
+			name:               "invalid listener",
+		},
+		{
+			route: createNormalRoute(gw),
+			gateway: &Gateway{
+				Source: gw,
+				Valid:  true,
+				Listeners: []*Listener{
+					createModifiedListener("listener-443", func(l *Listener) {
+						l.Source.Hostname = (*gatewayv1.Hostname)(helpers.GetPointer("*.example.org"))
+						helpers.GetPointer("*.example.com")
+					}),
+				},
+			},
+			expectedSectionNameRefs: []ParentRef{
+				{
+					Idx:         0,
+					Gateway:     client.ObjectKeyFromObject(gw),
+					SectionName: tr.Spec.ParentRefs[0].SectionName,
+					Attachment: &ParentRefAttachmentStatus{
+						AcceptedHostnames: map[string][]string{},
+						FailedCondition:   staticConds.NewRouteNoMatchingListenerHostname(),
+					},
+				},
+			},
+			expectedGatewayListeners: []*Listener{
+				createModifiedListener("listener-443", func(l *Listener) {
+					l.Source.Hostname = (*gatewayv1.Hostname)(helpers.GetPointer("*.example.org"))
+					helpers.GetPointer("*.example.com")
+				}),
+			},
+			name: "hostname does not match",
+		},
```

Comment:
I think we are missing a couple cases:
- nil section name
- empty section name
- listener does not exist
- invalid attachable route
- route kind not allowed

Reply:
empty section name means section name that is just empty quotes right?

Reply:
yep!